File: src/main/java/com/fwdekker/randomness/decimal/DecimalGroupAction.java
package com.fwdekker.randomness.decimal;

import com.fwdekker.randomness.DataInsertAction.ArrayAction;
import com.fwdekker.randomness.DataGroupAction;
import com.fwdekker.randomness.DataInsertAction;
import com.fwdekker.randomness.SettingsAction;


/**
 * All actions related to inserting decimals.
 */
public final class DecimalGroupAction extends DataGroupAction {
    @Override
    protected DataInsertAction getInsertAction() {
        return new DecimalInsertAction();
    }

    @Override
    protected DataInsertAction.ArrayAction getInsertArrayAction() {
        return new DecimalInsertAction().new ArrayAction();
    }

    @Override
    protected SettingsAction getSettingsAction() {
        return new DecimalSettingsAction();
    }
}

File: src/main/kotlin/com/fwdekker/randomness/decimal/DecimalGroupAction.kt
package com.fwdekker.randomness.decimal

import com.fwdekker.randomness.DataGroupAction


/**
 * All actions related to inserting decimals.
 */
class DecimalGroupAction : DataGroupAction() {
    override fun getInsertAction() = DecimalInsertAction()

    override fun getInsertArrayAction() = DecimalInsertAction().ArrayAction()

    override fun getSettingsAction() = DecimalSettingsAction()
}
